cdap-docs/application-templates/source/etl/templates/sources/batch/tpfsavro.rst

```diff
@@ -6,21 +6,76 @@
+For example, suppose there is an application that ingests data by writing to a TimePartitionedFileSet,
+where arrival time of the data is used as the partition key. You may want to create a pipeline that then
+reads newly arrived files, performs some data validation and cleansing, then writes to a Table.
+
+.. rubric:: Properties 
+
+**name:** Name of the TimePartitionedFileSet from which the records are to be read from.
+
+**schema:** The Avro schema of the record being read from the source as a JSON Object.
+
+**basePath:** Base path for the TimePartitionedFileSet. Defaults to the name of the
 dataset.
 
-**Duration:** Size of the time window to read with each run of the pipeline. The format is
-expected to be a number followed by a 's', 'm', 'h', or 'd' specifying the time unit, with
+**duration:** Size of the time window to read with each run of the pipeline. The format is
+expected to be a number followed by an 's', 'm', 'h', or 'd' specifying the time unit, with
 's' for seconds, 'm' for minutes, 'h' for hours, and 'd' for days. For example, a value of
 '5m' means each run of the pipeline will read 5 minutes of events from the TPFS source.
 
-**Delay:** Optional delay for reading from TPFS source. The value must be of the same
+**delay:** Optional delay for reading from TPFS source. The value must be of the same
 format as the duration value. For example, a duration of '5m' and a delay of '10m' means
-each run of the pipeline will read events from 15 minutes before its logical start time to
-10 minutes before its logical start time. The default value is 0.
+each run of the pipeline will read events 5 minutes of data from 15 minutes before its logical
+start time to 10 minutes before its logical start time. The default value is 0.
+
+.. rubric:: Example
+
+::
+
+  {
+    "name": "TPFSAvro",
+    "properties": {
+      "name": "webactivity",
+      "duration": "5m",
+      "delay": "1m",
+      "schema": "{
+        \"type\":\"record\",
+        \"name\":\"webactivity\",
+        \"fields\":[
+          {\"name\":\"date\",\"type\":\"string\"},
+          {\"name\":\"userid\",\"type\":\"long\"},
+          {\"name\":\"action\",\"type\":\"string\"},
+          {\"name\":\"item\",\"type\":\"string\"}
+        ]
+      }"
+    }
+  }
+
+This example reads from a TimePartitionedFileSet named 'webactivity', assuming the underlying
+files are in Avro format. TimePartitionedFileSets are partitioned by year, month, day, hour,
+and minute. Suppose the current run was scheduled to start at 10:00am on January 1, 2015.
+Since the 'delay' property is set to one minute, the most recent partition read will be one
+minute before the logical start time of the run. This corresponds to the partition for
+year 2015, month 1, day 1, hour 9, and minute 59. Since the 'duration' property is set to
+five minutes, a total of five partitions will be read. This means partitions for
+year 2015, month 1, day 1, hour 9, and minutes 55, 56, 57, 58, and 59 will be read. 
```

Comment:
Shouldn't it be 54, 55, 56, 57, and 58? Wouldn't then the last events be just before 59 minutes, and thus just before the delay? Otherwise, it doesn't match how the other works. You could use a 2m delay, though...